src/test/ui/existential_type2.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+
+#![feature(existential_type)]
+
+fn main() {}
+
+existential type Underconstrained<T: std::fmt::Debug>: 'static;
+//~^ ERROR `U` doesn't implement `std::fmt::Debug`
+
+// not a defining use, because it doesn't define *all* possible generics
```

Comment:
The span of the error here is pretty odd, no? I think we talked about this -- I was *expecting* to see errors arise from the WF checks on this function. I guess perhaps the WF definition for an existential type is not quite right...

Reply:
Yep, this code doesn't look right:

https://github.com/rust-lang/rust/blob/3d5753fda1ee8f729da1061e931e13b043f479a5/src/librustc/ty/wf.rs#L363-L367

We probably want to rewrite this to basically do the same as structs/enums:

https://github.com/rust-lang/rust/blob/3d5753fda1ee8f729da1061e931e13b043f479a5/src/librustc/ty/wf.rs#L292-L296

We *might* want to only do that for the case of explicit `existential type` declarations -- though I imagine it will be fine either way.

Reply:
Yea we need to skip it for `impl Trait`, otherwise https://github.com/rust-lang/rust/blob/master/src/test/ui/nll/ty-outlives/impl-trait-outlives.rs passes without errors

Reply:
Also

```rust
existential type WrongGeneric<T>: 'static;
//~^ ERROR the parameter type `T` may not live long enough

fn wrong_generic<T>(t: T) -> WrongGeneric<T> {
    t
}
```

stops being an error with NLL if I do that change.